src/ui/controller.js

```diff
@@ -0,0 +1,231 @@
+		 * 3. Setting {@link #ready} flag `true`.
+		 *
+		 * @returns {Promise} A Promise resolved when the initialization process is finished.
+		 */
+		init() {
+			if ( this.ready ) {
+				/**
+				 * This Controller already been initialized.
+				 *
+				 * @error ui-controller-init-reinit
+				 */
+				throw new CKEditorError( 'ui-controller-init-reinit: This Controller already been initialized.' );
+			}
+
+			return Promise.resolve()
+				.then( this._initView.bind( this ) )
+				.then( this._initCollections.bind( this ) )
+				.then( () => {
+					this.ready = true;
+				} );
+		}
+
+		/**
+		 * Adds a child controller to one of the {@link #collections}.
+		 * If this controller instance is ready, the child view will be initialized when added.
+		 * If this controller and child controller have views, the child view will be added
+		 * to corresponding region in this controller's view.
+		 *
+		 * @param {String} collectionName One of {@link #collections} the child should be added to.
+		 * @param {Controller} childController A child controller.
+		 * @param {Number} [index] Index at which the child will be added to the collection.
+		 * @returns {Promise} A Promise resolved when the child is added.
+		 */
+		addChild( collectionName, childController, index ) {
+			const collection = this.collections.get( collectionName );
+
+			// ChildController.init() returns Promise.
+			let promise = Promise.resolve();
+
+			collection.add( childController, index );
+
+			if ( this.ready ) {
+				if ( childController.view ) {
+					this.view.addChild( collectionName, childController.view, index );
+				}
+
+				if ( !childController.ready ) {
+					promise = promise.then( () => {
+						return childController.init();
+					} );
+				}
+			}
+
+			return promise;
+		}
+
+		/**
+		 * Removes a child controller from one of the {@link #collections}.
+		 * If this controller and child controller have views, the child view will be removed
+		 * from corresponding region in this controller's view.
+		 *
+		 * @param {String} collectionName One of {@link #collections} the child should be removed from.
+		 * @param {Controller} childController A child controller.
+		 * @returns {Controller} A child controller instance after removal.
+		 */
+		removeChild( collectionName, childController ) {
+			const collection = this.collections.get( collectionName );
+
+			collection.remove( childController );
+
+			if ( this.ready && childController.view ) {
+				this.view.removeChild( collectionName, childController.view );
+			}
+
+			return childController;
+		}
+
+		/**
+		 * Returns a child controller from one of the {@link #collections} at given `index`.
+		 *
+		 * @param {String} collectionName One of {@link #collections} the child should be retrieved from.
+		 * @param {Number} [index] An index of desired controller.
+		 * @returns {Controller} A child controller instance.
+		 */
+		getChild( collectionName, index ) {
```

Comment:
Seems to be unnecessary now as you can use `controller.collections.get( collectionName ).get( index )`.